#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
from os.path import abspath, basename, dirname, splitext

PACKAGE = basename(dirname(dirname(abspath(__file__))))

gen = ParameterGenerator()

gen.add("control_rate", double_t, 0, default=10.0, min=1.0, max=1000.0, description="Control rate in Hz")
gen.add("padding", double_t, 0, default=0.0, min=0.0, max= 10.0, description="Footprint inflation")
gen.add("passing_distance", double_t, 0, default=0.5, min=0.0, max= 10.0, description="Distance used to indicate that the part of the path is passed")
gen.add("enable_trajectory_validation", bool_t, 0, "Enable check of the local planner's output trajectory ", True)

exit(gen.generate(PACKAGE, PACKAGE + "_node", splitext(basename(__file__))[0]))
